bin/aptos

```diff
@@ -22,46 +26,65 @@ const getOS = () => {
   }
 };
 
+// Only works on Unix systems. This is fine because we only need to check for brew on
+// MacOS.
+const executableIsAvailable = (name) => {
+  try {
+    execSyncShell(`which ${name}`, { encoding: "utf8" });
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
 const main = async () => {
   const os = getOS();
   const binaryName = `${PNAME}-${VERSION}-${os}`;
 
-  // If binary does not exist, download it
-  if (!fs.existsSync(`${__dirname}/${binaryName}`)) {
-    await new Promise((resolve, reject) => {
-      const url = `https://github.com/aptos-labs/aptos-core/releases/download/${PNAME}-v${VERSION}/${PNAME}-${VERSION}-${os}-x86_64.zip`;
-      console.log(`Downloading aptos CLI version ${VERSION}`);
-      let downloadProcess;
-      if (os === "Windows") {
-        downloadProcess = spawn(
-          `curl -L -o C:/tmp/aptos.zip ${url} & powershell Expand-Archive -Path C:/tmp/aptos.zip -DestinationPath C:/tmp -Force & move C:\\tmp\\aptos.exe ${__dirname}/${binaryName}`,
-          {
-            stdio: "inherit",
-            shell: true,
-          }
-        );
-      } else {
-        downloadProcess = spawn(
-          `curl -L -o /tmp/aptos.zip ${url}; unzip -o -q /tmp/aptos.zip -d /tmp; mv /tmp/aptos ${__dirname}/${binaryName};`,
-          {
-            stdio: "inherit",
-            shell: true,
-          }
-        );
-      }
+  let path;
+  if (os === "MacOS") {
```

Comment:
can this be consolidate with the `if` block in line 71?

Reply:
I think this is cleaner, it lets me get the path first then install after.